src/firmware/guest/mod.rs

```diff
@@ -85,22 +86,21 @@ impl Firmware {
         &mut self,
         message_version: Option<u8>,
         ext_report_request: SnpExtReportReq,
-    ) -> Result<SnpReportRsp, Indeterminate<Error>> {
+    ) -> Result<(AttestationReport, CertTable), Indeterminate<Error>> {
```

Comment:
Eventually, A `Vec<UapiCertTableEntry>` is returned. But the `CertTableEntry` includes offset + pointer field, which is something we'd like to avoid presenting to a user (as Dov noted). We'd instead like to build a `Vec` of the Certificates themselves, and not worry about the other data in `CertTableEntry`.

Reply:
@tylerfanelli I think there is some confusion about which data type is being used here. The `CertTableEntry` in question is the definition provided in the UAPI level. It looks like this:

```rust 
/// Certificates which are accepted for [`CertTableEntry`]
pub enum SnpCertType {
    /// AMD Root Signing Key (ARK) certificate
    ARK,

    /// AMD SEV Signing Key (ASK) certificate
    ASK,

    /// Versioned Chip Endorsement Key (VCEK) certificate
    VCEK,

    /// Other (Specify GUID)
    OTHER(String),
}

/// An entry with information regarding a specific certificate.
pub struct CertTableEntry {
    /// A Specificy certificate type.
    cert_type: SnpCertType,

    /// The raw data of the certificate.
    data: Vec<u8>,
}

/// Certificates to send to the PSP.
pub struct CertTable {
    /// A vector of [`CertTableEntry`].
    pub entries: Vec<CertTableEntry>,
}
```

There is no offset or pointer field returned, and the `entries` field of the `CertTable` is the `Vec<CertTableEntry>` shown above.

Reply:
I'm sorry, you are right. I was mistakenly looking in `linux/host/types.rs`, my mistake.